### PR TITLE
RCJ-29: Create and save routes through drift

### PR DESCRIPTION
**Pull request by NicholasDawson**
## Description

- Made add route page save a route object to the database
- Made the routes page show routes from the database
- Removed an unnecessary unit test for tags

## Ticket

RCJ-29- [LINK](https://nmd2117.atlassian.net/browse/RCJ-29)

### CODE REVIEW

File: lib/add_route/add_page.dart [hunk 1]
```diff
@@ -116,35 +92,43 @@ class _AddPageState extends State<AddPage> {
...
 
-                Navigator.pop(context);
+                for (var tag in _selectedTags) {
+                  db.routeTagDao.insert(RouteTagCompanion(
+                    routeId: drift.Value(insertedRoute.id),
+                    tagId: drift.Value(tag.index)
+                  ));
+                }
+
+                if (!mounted) return;
```
Comment (devkav) [inline lib/add_route/add_page.dart:130]: Nitpick: always use curly braces with if statement

File: lib/routes/routes_page.dart [hunk 2]
```diff
@@ -356,14 +58,32 @@ class _RoutesPageBodyState extends State<_RoutesPageBody> {
...
                 builder: (context, filterChangeNotifier, child) {
-              return Column(
-                  children: routes
-                      .where((route) => filterChangeNotifier.filters
-                          .every((filter) => filter.isValid(route)))
-                      .map((ClimbingRoute route) =>
-                          RouteListElement(route: route))
-                      .toList());
-            }))
+              return StreamBuilder<List<RouteData>>(
```
Comment (devkav) [inline lib/routes/routes_page.dart]: Nitpick: Use shorthand return instead of {return ...}

File: lib/routes/routes_page.dart [hunk 3]
```diff
@@ -356,14 +58,32 @@ class _RoutesPageBodyState extends State<_RoutesPageBody> {
...
+                builder: (BuildContext context,
+                    AsyncSnapshot<List<RouteData>> snapshot) {
+                  if (snapshot.hasError) {
+                    return const Text("Error getting routes.");
+                  } else {
+                    switch (snapshot.connectionState) {
+                      case ConnectionState.none:
+                        return const CircularProgressIndicator();
+                      case ConnectionState.waiting:
+                        return const CircularProgressIndicator();
```
Comment (devkav) [inline lib/routes/routes_page.dart]: Have you tested this scenario to confirm if it looks good visually?

**Review by devkav — CHANGES_REQUESTED** [review]

File: lib/add_route/add_page.dart (hunk 1 above)
Comment (NicholasDawson) [reply lib/add_route/add_page.dart:130]: I think this way is more concise and is recommended by the dart linter [here](https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html)

File: lib/routes/routes_page.dart (hunk 2 above)
Comment (NicholasDawson) [reply lib/routes/routes_page.dart]: Okay I will do that.

File: lib/routes/routes_page.dart (hunk 3 above)
Comment (NicholasDawson) [reply lib/routes/routes_page.dart]: Yeah, it looks good. I attached a gif so you can let me know what you think.
studio64_6RfGogRKPC

Comment (devkav) [reply lib/routes/routes_page.dart]: Could you give this like 20 padding on the top?

File: lib/routes/routes_page.dart (hunk 2 above)
Comment (devkav) [reply lib/routes/routes_page.dart]: I am referring to the return statement, not the if statement

File: lib/routes/routes_page.dart (hunk 3 above)
Comment (NicholasDawson) [reply lib/routes/routes_page.dart]: Also, what do you think about the error message, I really don't anticipate any errors, but I think it looks good enough to move forward. 
image

**Review by devkav — COMMENTED** [review]
lgtm

**Review by devkav — APPROVED** [review]
lgtm

**Pull request merged**